File: apps/android/app/src/main/java/com/example/shop/core/model/PaginatedResult.kt
package com.example.shop.core.model

import kotlinx.serialization.Serializable

@Serializable
data class PaginatedResult<T>(
    val items: List<T>,
    val pagination: Pagination,
)

@Serializable
data class Pagination(
    val page: Int,
    val pageSize: Int,
    val total: Int,
    val totalPages: Int,
)

File: apps/ios/Shop/Core/Model/PaginatedResult.swift
import Foundation

struct PaginatedResult<T: Decodable & Sendable>: Decodable, Sendable {
    let items: [T]
    let pagination: Pagination
}

struct Pagination: Codable, Equatable, Sendable {
    let page: Int
    let pageSize: Int
    let total: Int
    let totalPages: Int
}
